File: work/deprecated/room/entity/ChineseCityEntity.java
package wangdaye.com.geometricweather.room.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import wangdaye.com.geometricweather.basic.models.ChineseCity;

/**
 * Chinese city entity.
 *
 * {@link ChineseCity}.
 * */

@Entity(tableName = "CHINESE_CITY_ENTITY")
public class ChineseCityEntity {

    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "_id") public Long id;

    @ColumnInfo(name = "CITY_ID") public String cityId;
    @ColumnInfo(name = "PROVINCE") public String province;
    @ColumnInfo(name = "CITY") public String city;
    @ColumnInfo(name = "DISTRICT") public String district;
    @ColumnInfo(name = "LATITUDE") public String latitude;
    @ColumnInfo(name = "LONGITUDE") public String longitude;
}

File: app/src/main/java/org/breezyweather/db/entities/ChineseCityEntity.kt
package org.breezyweather.db.entities

import io.objectbox.annotation.Entity
import io.objectbox.annotation.Id

/**
 * Chinese city entity.
 *
 * [ChineseCity].
 */
@Entity
data class ChineseCityEntity(
    @field:Id var id: Long = 0,
    var cityId: String,
    var province: String,
    var city: String,
    var district: String,
    var latitude: String,
    var longitude: String
)
